src/util/artifacts.js

```diff
@@ -14,12 +16,36 @@ export async function prepareArtifactsDir(
     const stats = await fs.stat(artifactsDir);
     if (!stats.isDirectory()) {
       throw new UsageError(
-        `--artifacts-dir=${artifactsDir} (this value is not a directory)`);
+        `"${artifactsDir}" exists and it is not a directory.`);
+    }
+    // If the artifactsDir already exists, check that we have the write permissions on it.
+    try {
+      await fs.access(artifactsDir, fs.W_OK);
+    } catch (accessErr) {
+      if (isErrorWithCode('EACCES', accessErr)) {
+        throw new UsageError(
+          `"${artifactsDir}" exists but the user lacks permissions on it.`);
```

Comment:
I think a prefix would be helpful here too. I'd also suggest saying specifically that the file is not writable:

    `--artifacts-dir="${artifactsDir}" exists but the user lacks permissions to write in it.`